e2e/tests/integration/shell-next/loading.spec.js

```diff
@@ -2,15 +2,37 @@
 
 // TODO before each slow requests to remoteEntry.js 1 sec
 describe("Nextjs shell: loading", () => {
+  beforeEach(() => {
+    // TODO - figure out a way to do this that doesn't introduce a race
+    cy.intercept("**/remoteEntry.js", (req) => {
```

Comment:
Maybe we can avoid the race condition by throwing an error differently. e.g. if we try to load a micro-frontend that doesn't exist:
```
<Host
  remote={{ packageName: "I-ll-throw-an-error-because-I-dont-exist-👹" }}
/>
```

Reply:
I'm not quite following how this would help the race condition here? Could you expand on what you were thinking? I was picturing having a promise that we can resolve which continues the test, rather than relying on an `N` second delay...

Reply:
I don't think I understand how the race happens by looking at the code and I haven't run it. I don't know either how ` cy.intercept("**/remoteEntry.js", (req) => {` works. My assumption was that `intercept` creates the race that you mention and so throwing the error without mocking/ intercepting anything could solve the issue. But it's a shot in the dark 😅.

What I suggested to try is:
- Create a page http://localhost:44443/wrong-package-name
- In that page `<Host  remote={{ packageName: "I-ll-throw-an-error-because-I-dont-exist-👹" }} />`
- In the test `cy.visit("http://localhost:44443/wrong-package-name");` without `beforeEach(() => {`

If it doesn't solve the race then let's merge it and figure that out another time.

Reply:
Ah I'm with you - I haven't actually managed to prove it's a race, but we're basically asserting on a loading state that only exists for 1 second in the current implementation, so unless I'm missing something Cypress is doing, a slowness could cause issues here. For the sake of progress, I think you're right - lets merge this and have a look another time.